### PR TITLE
feat(bigtable/emulator): allow listening on Unix Domain Sockets

### PR DESCRIPTION
**Pull request by flokli**
cbtemulator listening on unix domain sockets is much easier than trying to allocate free TCP ports, especially if many cbtemulators are run at the same time in integration tests.

This adds an additional flag, address, which has priority if it's set, rather than host:port.

`NewServer` already takes a `laddr string`, so we simply check for it to contain slashes, and if so, listen on unix, rather than TCP.

### CODE REVIEW

**Comment by flokli** [comment]
This can be tested as follows:

```
$ ./emulator --address $PWD/emulator.sock &
Cloud Bigtable emulator running on /home/flokli/dev/google-cloud-go/bigtable/cmd/emulator/emulator.sock
$ export BIGTABLE_EMULATOR_HOST=unix://$PWD/emulator.sock
$ cbt -instance instance-1 -project project-1 createtable table-1
2024/03/29 11:02:04 -creds flag unset, will use gcloud credential
$ cbt -instance instance-1 -project project-1 ls
2024/03/29 11:02:15 -creds flag unset, will use gcloud credential
table-1
```

No changes in the client sdk itself needed, only in the emulator.

**Comment by flokli** [comment]
conventionalcommits.org seems stuck on an old commit message, not sure how to get it unstuck.

**Comment by flokli** [comment]
I got confused over the `conventionalcommits.org ` check - it was complaining about the PR *title*, not commit messages. It's green now, and the PR is rebased. Would be nice to get some feedback :-)

File: bigtable/bttest/inmem.go [hunk 1]
```diff
@@ -106,7 +107,15 @@ type server struct {
 // The Server will be listening for gRPC connections, without TLS,
 // on the provided address. The resolved address is named by the Addr field.
 func NewServer(laddr string, opt ...grpc.ServerOption) (*Server, error) {
-	l, err := net.Listen("tcp", laddr)
+	var l net.Listener
+	var err error
+
+	// If the address contains slashes, listen on a unix domain socket instead.
+	if strings.Contains(laddr, "/") {
```
[the remark this answers is not in the document]
Comment (bhshkh) [reply bigtable/bttest/inmem.go:116]: Please ignore this comment. I looked more into this and understand that the current approach is correct

**Review by bhshkh — APPROVED** [review]

**Pull request merged**